vllm/config.py

```diff
@@ -160,6 +160,9 @@ def _verify_quantization(self) -> None:
             is_format_marlin = (quant_cfg.get("checkpoint_format") == "marlin"
                                 or quant_cfg.get("is_marlin_format", False))
 
+            is_format_marlin_24 = (
```

Comment:
We should think about how to clean this up and not have this marlin specific code in `vllm/config.py`.

One way to do it that doesn't require more registries: Have an optional class variable `checkpoint_format` in the gptq compatible QuantizationConfig and then in this code, iterate  through `QUANTIZATION_METHODS` and see if one of them has the associated checkpoint format.

Reply:
Good suggestion. I have changed the code to encapsulate the marlin specific checkpoint checks into marlin config classes. Tell me if it looks good now.

Reply:
I'm thinking about something a little more radical like replacing this whole code block with
```
for name, method in QUANTIZATION_METHODS.values():
    if method.supports_checkpoint(quant_cfg):
        self.quantization = name
```

and you would have a default implementation of supports_checkpoint for QuantizationConfig that returns `False`, and Marlin would implement the method, print the appropriate warnings and return `True` if the quantization should be overridden.

That way you can remove all occurrences of marlin from the config.py file, and this mechanism can also be used by other quantization schemes :)

Reply:
I see, I can try it

Reply:
@pcmoritz I have redid the config.py part that you proposed to change. It looks cleaner now :)